src/graphics/device-cache.js

```diff
@@ -0,0 +1,50 @@
+/** @typedef {import('./graphics-device.js').GraphicsDevice} GraphicsDevice */
+
+/**
+ * A cache storing shared resources associated with a device. The resources are removed
+ * from the cache when the device is destroyed.
+ *
+ * @ignore
+ */
+class DeviceCache {
```

Comment:
Sorry I'm not crazy about this structure. Took me a while to figure out that `DeviceCache` instances contain a `data` member -and- a `map` of device -> 'DeviceCache' instance. So in some cases the map is used and data unused, in other cases visa versa.

I'd much prefer two interfaces: one for the device cache containing a map of devices and another for the thing that the device cache map stores.

Reply:
In fact, you could remove the data member and just store the entry directly.

Reply:
great suggestion, and I've done just that, thanks!